android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java

```diff
@@ -46,47 +58,53 @@ public static void promptToInstallWallet() {
     if (act == null) {
       return;
     }
+
+    final Intent appStoreIntent =
+        new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName));
+
     AlertDialog.Builder builder;
     builder = new AlertDialog.Builder(act);
-    builder.setTitle(R.string.wallet_missing);
-    builder.setMessage(act.getString(R.string.install_wallet_from_iab));
-    Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
+    builder.setTitle("APPC Wallet Missing");
+    builder.setMessage("To complete your purchase, you have to install an AppCoins wallet");
+    Log.d("String name: ", "To complete your purchase, you have to install an AppCoins wallet");
 
-    builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
+    builder.setPositiveButton("INSTALL", new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialog, int which) {
-        act.startActivity(
-            new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
+        OnInstallClickAction(act, appStoreIntent);
       }
     });
 
-    builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
+    builder.setNegativeButton("SKIP", new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialogInterface, int i) {
         dialogInterface.cancel();
       }
     });
 
     builder.setIcon(android.R.drawable.ic_dialog_alert);
+    Log.d(TAG, "promptToInstallWallet: builder.show()");
     builder.show();
   }
 
-  public static Activity getActivity() throws Exception {
-    Class activityThreadClass = Class.forName("android.app.ActivityThread");
-    Object activityThread = activityThreadClass.getMethod("currentActivityThread")
-        .invoke(null);
-    Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
-    activitiesField.setAccessible(true);
-    ArrayMap activities = (ArrayMap) activitiesField.get(activityThread);
-    for (Object activityRecord : activities.values()) {
-      Class activityRecordClass = activityRecord.getClass();
-      Field pausedField = activityRecordClass.getDeclaredField("paused");
-      pausedField.setAccessible(true);
-      if (!pausedField.getBoolean(activityRecord)) {
-        Field activityField = activityRecordClass.getDeclaredField("activity");
-        activityField.setAccessible(true);
-        Activity activity = (Activity) activityField.get(activityRecord);
-        return activity;
-      }
+  private static void OnInstallClickAction(Activity act, Intent appStoreIntent) {
+
+    //Check if the user has aptoide installed and open the aptoide's wallet page
+    if (hasAptoideInstalled()) {
+      Log.d("AptoideInstallation", "Aptoide is installed on this device");
+      appStoreIntent.setPackage(aptoidePackageName);
```

Comment:
There as been an update and additional details are required for the dialog that redirects to Aptoide.